rootfs/api/models/config.py

```diff
@@ -137,13 +111,12 @@ def save(self, **kwargs):
                         # error if unsetting non-existing key
                         if key not in data:
                             raise UnprocessableEntity('{} does not exist under {}'.format(key, attr))  # noqa
-
                         data.pop(key)
                     else:
                         data[key] = value
                 setattr(self, attr, data)
 
-            self.set_healthchecks()
+            self._migrate_legacy_healthcheck()
```

Comment:
I fixed this in 6fd0add by only injecting the port if it is not already defined (such as when using `deis config:set HEALTHCHECK_URL`). That should alleviate your concern with 

> What-if the client does all his config first and does a build next...then the health check might have wrong port